src/SFML/Window/WindowBase.cpp

```diff
@@ -0,0 +1,272 @@
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/WindowBase.h>
+#include <SFML/Window/WindowBaseStruct.h>
+#include <SFML/Internal.h>
+#include <SFML/Window/CursorStruct.h>
+#include <SFML/ConvertEvent.h>
+
+
+////////////////////////////////////////////////////////////
+sfWindowBase* sfWindowBase_create(sfVideoMode mode, const char* title, sfUint32 style)
+{
+    // Convert video mode
+    sf::VideoMode videoMode(mode.width, mode.height, mode.bitsPerPixel);
+
+    // Create the window
+    sfWindowBase* windowBase = new sfWindowBase;
+    windowBase->This.create(videoMode, title, style);
+
+    return windowBase;
+}
+
+
+////////////////////////////////////////////////////////////
+sfWindowBase* sfWindowBase_createUnicode(sfVideoMode mode, const sfUint32* title, sfUint32 style)
+{
+    // Convert video mode
+    sf::VideoMode videoMode(mode.width, mode.height, mode.bitsPerPixel);
+
+    // Create the window
+    sfWindowBase* windowBase = new sfWindowBase;
+    windowBase->This.create(videoMode, title, style);
+
+    return windowBase;
+}
+
+
+////////////////////////////////////////////////////////////
+sfWindowBase* sfWindowBase_createFromHandle(sfWindowHandle handle)
+{
+    // Create the window
+    sfWindowBase* windowBase = new sfWindowBase;
+    windowBase->This.create(handle);
+
+    return windowBase;
+}
+
+
+////////////////////////////////////////////////////////////
+void sfWindowBase_destroy(sfWindowBase* windowBase)
+{
+    delete windowBase;
+}
+
+
+////////////////////////////////////////////////////////////
+void sfWindowBase_close(sfWindowBase* windowBase)
+{
+    CSFML_CALL(windowBase, close());
+}
+
+
+////////////////////////////////////////////////////////////
+sfBool sfWindowBase_isOpen(const sfWindowBase* windowBase)
+{
+    CSFML_CALL_RETURN(windowBase, isOpen(), sfFalse);
+}
+
+
+////////////////////////////////////////////////////////////
+sfBool sfWindowBase_pollEvent(sfWindowBase* windowBase, sfEvent* event)
+{
+    CSFML_CHECK_RETURN(windowBase, sfFalse);
+    CSFML_CHECK_RETURN(event, sfFalse);
+
+    // Get the event
+    sf::Event SFMLEvent;
```

Comment:
```suggestion
    sf::Event sfmlEvent;
```

I know this is just copy-pasted